### PR TITLE
Fixed Shell.NavBarHasShadow on Android

### LINKED ISSUE #19458: Shell.NavBarHasShadow not working correctly on Android
### Description

`Shell.NavBarHasShadow` does not show the shadow under title bar on Android when being set on `ContentPage`. Stranglely when I debug the app and set this to false, then to true, the shadow does display.

### Steps to Reproduce

1. Create new .NET MAUI App, version .NET 8, or clone repo here: [https://github.com/gubhit1s/MauiApp1](url)
2. In `AppShell.xaml`, set `Shell.FlyoutBehavior="Flyout"`
3. In MainPage.xaml, add `Shell.NavBarHasShadow="True"`
4. Debug the project, the shadow doesn't appear.
image

5. In debug mode, change `Shell.NavBarHasShadow="False"`, the shadow doesn't appear.
6. Also in debug mode, change `Shell.NavBarHasShadow="True"`, the shadow appears.
image

### Link to public reproduction project repository

https://github.com/gubhit1s/MauiApp1

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 14

### Did you find any workaround?

Trying to switch to CustomRenderer

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Description of Change

Fixed Shell.NavBarHasShadow on Android

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/19458

https://github.com/dotnet/maui/assets/42434498/cebd0115-cca5-4e6e-97b7-bda17bb930f5

### CODE REVIEW

File: src/Controls/samples/Controls.Sample/Pages/Core/ShellGalleries/ShellChromeGallery.cs [hunk 1]
```diff
@@ -101,6 +101,11 @@ void OnToggleFlyoutBackgroundColor(object sender, EventArgs e)
 			flyoutBackgroundColor.Background = AppShell.FlyoutBackground;
 		}
 
+		void OnToggleNavBarHasShadow(object sender, EventArgs e)
+		{
+			Shell.SetNavBarHasShadow(this, !Shell.GetNavBarHasShadow(this));
```
Comment (jsuarezruiz) [inline src/Controls/samples/Controls.Sample/Pages/Core/ShellGalleries/ShellChromeGallery.cs:106]: Updated the Shell sample to easily test the `NavBarHasShadow` property.

File: src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Android.cs [hunk 2]
```diff
@@ -28,6 +28,40 @@ namespace Microsoft.Maui.DeviceTests
 	[Category(TestCategory.Shell)]
 	public partial class ShellTests
 	{
+		[Theory]
+		[InlineData(true)]
+		[InlineData(false)]
+		public async Task CanHideNavBarShadow(bool navBarHasShadow)
```
Comment (jsuarezruiz) [inline src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Android.cs:34]: Added Device Tests.

**Review by jsuarezruiz — DISMISSED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellToolbarTracker.cs [hunk 3]
```diff
@@ -561,15 +561,15 @@ void UpdateNavBarHasShadow(Page page)
 			if (page == null || !_appBar.IsAlive())
 				return;
 
+			// 4 is the default
+			_appBarElevation = _appBar.Context.ToPixels(4);
```
Comment (PureWeen) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellToolbarTracker.cs:565]: The point of the original code was to reduce the amount of calls to `_appBar.Context`. The original code did a really bad job of it :-) 

Maybe something like this? I think this is what the original author was intending.

```C#
			if (Shell.GetNavBarHasShadow(page))
			{
				if (_appBarElevation <= 0)
				     _appBarElevation = _appBar.Context.ToPixels(4);

				_appBar.SetElevation(_appBarElevation);
			}
			else
			{
				// 4 is the default
				_appBar.SetElevation(0f);
			}
```

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellToolbarTracker.cs (hunk 3 above)
Comment (kubaflo) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellToolbarTracker.cs]: Changed, thanks!

**Comment by PureWeen** [comment]
Failing tests unrelated

**Pull request merged**